pkg/models/app.go

```diff
@@ -12,3 +12,10 @@ type App struct {
 	DeployedVersions      *[]Version `json:"deployedVersions,omitempty"`
 	DeletedAt             string     `json:"deletedAt,omitempty"`
 }
+
+func NewAppByNameAppID(name, appId string) *App {
```

Comment:
Suggestion:

```go
// NewApp returns a new instance of App
func NewApp(name string, appID string, deployedVersions *[]Version) *App {
	return &App{
		ID:               uuid.New().String(),
		AppName:          name,
		AppID:            appID,
		DeployedVersions: deployedVersions,
	}
}
```

No need for a specific constructor by app ID and app name in my opinion You can just add one extra parameter and pass nil. Also, it is more common to use composite literals over `new`. This is suggested in the official docs: [composite literals](https://golang.org/doc/effective_go.html#composite_literals).

Reply:
Hi @craicoverflow,

According to the business no make sense pass the other attributes. The only information that should be used to create the app is the name and the appId. The UUID should be created by the Server when it will be added. 

Also, note that one class can have many constructors as it is required and because of this and because of the business roles of this object I do not agree with your suggestion.

Reply:
If all that is needed to create an app is name and appID, you could just call it `NewApp`.

Reply:
Okay, well I still think you should return a composite literal.

Reply:
It is returning a composite literal. "We can simplify it using a composite literal, which is an expression that creates a new instance each time it is evaluated."

It will create a new instance each time it is evaluated based on the only info that can be provided to do it. The UUID should not be created in this case. I don't want to add the UUID and the UUID is created when the POST REST is called in its respective service implementation.

Reply:
Please read the documentation I sourced. It literally explains the difference between a composite literal:

> There's a lot of boiler plate in there. We can simplify it using a composite literal, which is an expression that creates a new instance each time it is evaluated. I am not talking about the UUID, that is fine.

```go
func NewFile(fd int, name string) *File {
    if fd < 0 {
        return nil
    }
    f := File{fd, name, nil, 0}
    return &f
}
```